File: 20220907/모범답안/20220907_08.js/trritorial.js
function search(arr, Countries, Country, count)
{

  for(var i = 0; i < Country.length; i++)
  {
    if(Country[i] == 1 && arr[i] == 0)
    {
      arr[i] = 1;
      count += 1;
      count = search(arr, Countries, Countries[i], count);
    }
  }

  return count;
}

function solution(A)
{
  var count = 0;
  var result = 0;
  var arr = new Array(A.length);

  arr.fill(0);

  for(var i = 0; i < A.length; i++)
  {
    if(arr[i] == 0)
    {
      count = 1;
      arr[i] = 1

      count = search(arr, A, A[i], count);

      if(count > result)
      {
        result = count;
      }
    }
  }

  return result;
}

File: 20220907/모범답안/20220907_08.js/trritorial.py
def search(arr, Countries, Country, count):
    for i , country in enumerate(Country):
        if country == 1 and arr[i] == 0:
            arr[i] = 1
            count += 1
            count = search(arr, Countries, Countries[i], count)
    return count

def solution(A):
    count = 0
    result = 0
    arr = [0 for i in range(len(A))]

    for i in range(len(A)):
        if arr[i] == 0:
            count = 1
            arr[i] = 1
            count = search(arr, A, A[i], count)
            if count > result:
                result = count

    return result